cmd/utils/flags.go

```diff
@@ -672,6 +672,17 @@ var (
 		Value: eth.DefaultConfig.Istanbul.LookbackWindow,
 	}
 
+	// Announce settings
+	AnnounceGossipPeriodFlag = cli.Uint64Flag{
+		Name:  "announce.gossipperiod",
+		Usage: "Time duration (in seconds) between gossiped announce messages",
+		Value: eth.DefaultConfig.Istanbul.AnnounceGossipPeriod,
+	}
+	AnnounceAggressiveGossipOnEnablementFlag = cli.BoolFlag{
+		Name:  "announce.aggressivegossiponenablement",
+		Usage: "Specifies if this node should do aggressive gossip on announce enablement",
+	}
+
```

Comment:
I actually argue that is it not a detail of Istanbul, and that it just happens that all the code is in the istanbul package.   And that the announce protocol is a layer underneath the istanbul consensus layer.

E.g. none of the announce data structes and messages are dependent on any of the consensus message.

wdty?

Reply:
If a node were running PoW, would it ever use the announce messages? No, right or am I missing something? Essentially I would lump it in with Istanbul because it's only relevant if you are running Istanbul. (In my view flag naming is communicating functionality groups to the reader moreso than the way the code is structured)

Reply:
That is correct.  However, PoW may not be a great example w.r.t. my argument, as there is no "valset", since any node could be a "validator" without an election.

However, it wouldn't be too hard to get the announce protocol to work for geth's PoA implementation.  Or any other consensus algorithm that requires a subnetwork among a designated set of nodes.  So my thinking that "in principle" it is independent of the consensus algorithm, but it just happens to be currently tightly coupled with it.